### ISSUE TITLE
With backward method of CUDA Variables, grad is None if gradient is supplied

### ISSUE THREAD

**Issue by fvncc** [pytorch/pytorch#2730]
Basically, this prints None:

```
x = Variable(torch.ones(10), requires_grad=True).cuda()
y = x * Variable(torch.linspace(1, 10, 10), requires_grad=False).cuda()
y.backward(torch.ones(10).cuda())
print(x.grad)
```

**Comment by ruotianluo** [comment]
try this:
```
_x = Variable(torch.ones(10), requires_grad=True)
x = _x.cuda()
y = x * Variable(torch.linspace(1, 10, 10), requires_grad=False).cuda()
y.backward(torch.ones(10).cuda())
print(x.grad)
print(_x.grad)
```
This is because x is an intermediate node.

**Comment by fvncc** [comment]
Ah I see now, thanks :)
